### PR TITLE
feat: add rust for rustaceans as a resource in docs

### PR DESCRIPTION
**Pull request by joshuajbouw**
Rust for Rustaceans is one of my favourite resources for Rust and introduced me to great idiomatic programming techniques which take you to the next level towards mastering Rust. It should be mentioned in the documentation.

## Synopsis

For developers who’ve mastered the basics, this book is the next step on your way to professional-level programming in Rust. It covers everything you need to build and maintain larger code bases, write powerful and flexible applications and libraries, and confidently expand the scope and complexity of your projects.

## Author
 
Author Jon Gjengset takes you deep into the Rust programming language, dissecting core topics like ownership, traits, concurrency, and unsafe code. You’ll explore key concepts like type layout and trait coherence, delve into the inner workings of concurrent programming and asynchrony with async/await, and take a tour of the world of no_std programming. Gjengset also provides expert guidance on API design, testing strategies, and error handling, and will help develop your understanding of foreign function interfaces, object safety, procedural macros, and much more.

### CODE REVIEW

File: docs/practices/rust.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ Alternatives are:
 * [Programming
   Rust](https://www.amazon.com/Programming-Rust-Fast-Systems-Development/dp/1491927283)
   is an alternative book which moves a bit faster.
+* [Rust for Rustaceans: Idiomatic Programming for Experienced Developers](https://a.co/d/i7zIu3e) excellent book that serves as a next step after becoming intermediate with the language.
```
Comment (nikurt) [inline docs/practices/rust.md:33]:
```suggestion
* [Rust for Rustaceans: Idiomatic Programming for Experienced Developers](https://a.co/d/i7zIu3e)
  excellent book that serves as a next step after becoming intermediate with the language.
```

**Review by nikurt — APPROVED** [review]

**Comment by matklad** [comment]
This is already included in the mastery section: https://near.github.io/nearcore/practices/rust.html#language-mastery

**Comment by matklad** [comment]
though probably makes sense to put it as the first entry there

**Comment by matklad** [comment]
cc https://github.com/near/nearcore/pull/8046

**Comment by joshuajbouw** [comment]
Oops! Totally missed it there.

**Pull request closed without merging**